clang/lib/DPCT/DPCT.cpp

```diff
@@ -912,23 +890,24 @@ int runDPCT(int argc, const char **argv) {
     EndPos = SourceCode.find_last_of('\n', EndPos);
     QueryAPIMappingSrc =
         SourceCode.substr(StartPos, EndPos - StartPos + 1).str();
-#ifdef _WIN32
-    if (win_unsupported) {
-      llvm::outs() << "CUDA API:" << llvm::raw_ostream::GREEN
-                   << QueryAPIMappingSrc << llvm::raw_ostream::RESET
-                   << "The API mapping query for this API is not available "
-                    "on Windows.\n";
-      return MigrationSucceeded;
-    }
-#else
-    if (lin_unsupported) {
+    // Print static migration info for cudaGraphicsD3D11RegisterResource
+    // on linux, as the API is unavailable on Linux and hence, no API mapping.
+#ifndef _WIN32
+    auto CudaAPI = SourceCode.substr(
+        StartPos + 1, StringRef(QueryAPIMappingSrc).find_first_of('(') - 1);
+    CudaAPI = CudaAPI.trim(' ');
+    if (CudaAPI == "cudaGraphicsD3D11RegisterResource") {
```

Comment:
Can we just use `SourceCode.contain("cudaGraphicsD3D11RegisterResource")` to simplify the logic for this case? because no other API example will contain this API.